docs/source/building/variants.rst

```diff
@@ -0,0 +1,676 @@
+3. a file named ``conda_build_config.yaml`` in the same folder as ``meta.yaml``
+   with your recipe
+4. Any additional files specified on the command line with the
+   ``--variant-config-files`` or ``-m`` command line flags, which can be passed
+   multiple times for multiple files. The ``conda build`` and ``conda render``
+   commands accept these arguments.
+
+Files found later in this search order clobber the values from earlier files.
+
+
+Using variants with the conda-build API
+---------------------------------------
+
+
+Ultimately, a variant is just a dictionary. This dictionary is provided directly
+to Jinja2 - you can use any declared key from your variant configuration in your
+Jinja2 templates. There are two ways that you can feed this information into the
+API:
+
+1. pass the ``variants`` keyword argument to API functions. Currently, the
+   ``build``, ``render``, ``get_output_file_path``, and ``check`` functions
+   accept this argument. ``variants`` should be a dictionary with values being
+   lists of versions to iterate over. These are aggregated as detailed in the
+   Aggregation of multiple variants section below.
+
+2. Set the ``variant`` member of a Config object. This is just a simple
+   dictionary. The values for fields should be strings, except "extended keys",
+   which are documented in the `Extended keys`_ section below.
+
+
+Again, with meta.yaml contents like:
+
+.. code-block:: yaml
+
+   package:
+       name: compiled-code
+       version: 1.0
+
+   requirements:
+       build:
+           - python {{ python }}
+       run:
+           - python {{ python }}
+
+You could supply a variant to build this recipe like so:
+
+.. code-block:: python
+
+   variants = {'python': ['2.7', '3.5']}
+   api.build(path_to_recipe, variants=variants)
+
+
+Special variant keys
+--------------------
+
+
+* ``pin_run_as_build``: should be a dictionary. Keys are package names. Values
+  are "pinning expressions" - explained in more detail in `Customizing
+  compatibility`_. This is a generalization of the ``numpy x.x`` spec.
+* ``extend_keys``: specifies keys that should be aggregated, rather than
+  clobbered, by later variants. These are detailed below in the `Extended keys`_
+  section.
+* ``runtimes``: detailed further in `Extra Jinja2 functions`_.
+* ``exclude_from_build_hash``: list of package names to exclude from
+  requirements/build when computing hash. Described further in `Avoiding
+  unnecessary builds`_.
+
+
+Avoiding unnecessary builds
+---------------------------
+
+
+To avoid building variants of packages where pinning does not necessitate having
+different builds, you can use the ``exclude_from_build_hash`` key in your
+variant. The way this works is that all variants are evaluated, but if any
+hashes are the same, then they are considered duplicates, and are deduplicated.
+By omitting some packages from the build dependencies, we can avoid creating
+unnecessarily unique hashes, and allow this deduplication.
```

Comment:
I would probably word this "avoid creating unnecessarily specific hashes".  Since hashes are meant to be unique, it was a little confusing to understand at first.